pkg/kubernetes/deploy.js

```diff
@@ -0,0 +1,373 @@
+        $('#deploy-app-namespace-field-custom').parent().removeClass('has-error');
+        $('#deploy-app-manifest-file').parent().removeClass('has-error');
+
+    }
+
+    function pre_init() {
+        //alert("pre_init")
+        var firstTime = true;
+        var dlg = $('#deploy-app-dialog');
+        var btn = $('#deploy-app-start');
+        var manifest_file = $('#deploy-app-manifest-file');
+        var manifest_file_note = $('#deploy-app-manifest-file-note');
+        var manifest_file_details = $("#deploy-app-manifest-file-note-details");
+        var ns_selector = $('#deploy-app-namespace-field');
+        appdeployer.jsondata = "";
+        var text = "";
+
+        btn.on('click', function() {
+            deploy_app();
+        });
+
+        dlg.on('show.bs.modal', function() {
+            //alert("show.bs.models");
+            if (firstTime) {
+                var optionls = [];
+                var nslist = get_kentities("namespaces");
+                for (var i = 0; i < nslist.length; i++) {
+                    optionls.push('<option translatable="yes" value="' + nslist[i] + '">' + nslist[i] + '</option>');
+                }
+                optionls.push('<option translatable="yes" value="Custom Namespace">Custom Namespace</option>');
+                var optionlshtml = optionls.join('');
+                ns_selector.prepend(optionlshtml);
+                ns_selector.selectpicker('refresh');
+                firstTime = false;
+            }
+            manifest_file.val("");
+            deploy_dialog_remove_errors();
+        });
+
+
+        dlg.on('keypress', function(e) {
+            if (e.keyCode === 13)
+                btn.trigger('click');
+        });
+
+        manifest_file.on('change', function() {
+            //alert("manifest_file")
+
+            manifest_file_note.hide();
+            manifest_file_details.hide();
+            manifest_file.parent().removeClass('has-error');
+
+            var files, file, reader;
+            files = manifest_file[0].files;
+            if (files.length != 1) {
+                text = "No json File was selected.Please select a json file. ";
+                manifest_file_note.show();
+                manifest_file_details.text(text);
+                manifest_file_details.show();
+                manifest_file.parent().addClass('has-error');
+                return;
+            }
+            file = files[0];
+            if (!file.type.match("json.*")) {
+                text = "Selected file is Not a Json file.Please select a json file. ";
+                manifest_file_note.show();
+                manifest_file_details.text(text);
+                manifest_file_details.show();
+                manifest_file.parent().addClass('has-error');
+                return;
+            }
+            reader = new window.FileReader();
+            reader.onerror = function() {
+                text = "Unable to Read the file.Please check the json file. ";
```

Comment:
Can we provide more details about the error? Is ```reader.onerror``` called with an exception argument?